CDP4IME/CDP4IMEBootstrapper.cs

```diff
@@ -61,31 +65,66 @@ protected override DependencyObject CreateShell()
         /// </summary>
         protected override void InitializeShell()
         {
+            var appSettings = this.Container.GetExportedValue<IAppSettingsService>();
+            var appServiceSettings = this.ReadAppSettings(appSettings);
+
             this.UpdateBootstrapperState("Loading CDP4 Plugins");
             var pluginCatalog = new CDP4PluginLoader();
             foreach (var directoryCatalog in pluginCatalog.DirectoryCatalogues)
             {
-                this.AggregateCatalog.Catalogs.Add(directoryCatalog);
+                var fileName = Path.GetFileName(directoryCatalog.FullPath);
+
+                if ((fileName != null) && !appServiceSettings.DisabledPlugins.Select(x => x.ToUpper()).ToList().Contains(fileName.ToUpper()))
+                {
+                    this.AggregateCatalog.Catalogs.Add(directoryCatalog);
 
-                this.UpdateBootstrapperState(string.Format("DirectoryCatalogue {0} Loaded", directoryCatalog.FullPath));
+                    this.UpdateBootstrapperState($"DirectoryCatalogue {directoryCatalog.FullPath} Loaded");
+                }
             }
 
-            this.UpdateBootstrapperState(string.Format("{0} CDP4 Plugins Loaded", pluginCatalog.DirectoryCatalogues.Count));
+            this.UpdateBootstrapperState($"{pluginCatalog.DirectoryCatalogues.Count} CDP4 Plugins Loaded");
 
             this.UpdateBootstrapperState("Initializing the Shell");
             base.InitializeShell();
             var shell = (Shell)this.Shell;
-            var dialogNavigationService = Container.GetExportedValue<IDialogNavigationService>();
-            shell.DataContext = new ShellViewModel(dialogNavigationService);
+            var dialogNavigationService = this.Container.GetExportedValue<IDialogNavigationService>();
+
+            shell.DataContext = new ShellViewModel(dialogNavigationService, appSettings);
 
             this.UpdateBootstrapperState("Setting up Regions");
             var regionmanager = this.Container.GetExportedValue<IRegionManager>();
             foreach (var region in regionmanager.Regions)
             {
-                this.UpdateBootstrapperState(string.Format("Loaded Region: {0} ", region.Name));
+                this.UpdateBootstrapperState($"Loaded Region: {region.Name} ");
             }
 
-            Application.Current.MainWindow = shell;            
+            Application.Current.MainWindow = shell;
+        }
+
+        /// <summary>
+        /// Reads the app settings from disk
+        /// </summary>
+        internal ImeAppSettings ReadAppSettings(IAppSettingsService appSettingService)
```

Comment:
I actually do not see why the settings service cannot perform the inside itself? The settings should be loaded into mem when it is initialized.

Reply:
Settings are being loaded within the class itself. Code has been refactored